models/Import/Service/QtiTestItemsService.php

```diff
@@ -0,0 +1,105 @@
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoResultServer\models\Import\Service;
+
+use oat\taoDeliveryRdf\model\DeliveryContainerService;
+use oat\taoProctoring\model\execution\DeliveryExecutionManagerService;
+use oat\taoQtiTest\models\runner\QtiRunnerService;
+use oat\taoQtiTest\models\runner\QtiRunnerServiceContext;
+use qtism\data\AssessmentTest;
+use qtism\data\ExtendedAssessmentItemRef;
+use qtism\data\ExtendedAssessmentSection;
+use qtism\data\TestPart;
+use taoQtiTest_helpers_Utils;
+
+class QtiTestItemsService
+{
+    private QtiRunnerService $qtiRunnerService;
+    private DeliveryExecutionManagerService $deliveryExecutionService;
+    private DeliveryContainerService $deliveryContainerService;
+
+    public function __construct(
+        QtiRunnerService $qtiRunnerService,
+        DeliveryExecutionManagerService $deliveryExecutionService,
+        DeliveryContainerService $deliveryContainerService,
+    ) {
+
+        $this->qtiRunnerService = $qtiRunnerService;
+        $this->deliveryExecutionService = $deliveryExecutionService;
+        $this->deliveryContainerService = $deliveryContainerService;
+    }
+
+    public function getItemsByDeliveryExecutionId(string $deliveryExecutionId): array
+    {
+        $context = $this->getServiceContext($deliveryExecutionId);
+        $testDefinition = $this->getDefinition($context);
+        $testParts = [];
+        /** @var TestPart $testPart */
+        foreach ($testDefinition->getTestParts() as $testPart) {
+            $sections = [];
+            /** @var ExtendedAssessmentSection $section */
+            foreach ($testPart->getAssessmentSections() as $section) {
+                $items = [];
+                /** @var ExtendedAssessmentItemRef $item */
+                foreach ($section->getSectionParts() as $item) {
+                    $itemData = $this->qtiRunnerService->getItemData($context, $item->getHref());
+                    $itemData['data']['isExternallyScored'] = $this->isExternallyScored($itemData['data'] ?? []);
+
+                    $items[$item->getIdentifier()] = $itemData['data'];
+                }
+                $sectionId = $section->getIdentifier();
+                $sections[$sectionId] = $items;
+            }
+            $testPartId = $testPart->getIdentifier();
+            $testParts[$testPartId] = $sections;
+        }
+
+        return $testParts;
```

Comment:
suggest flat array with attributes as return for this:
```
[
  [
     "externalScored" => '',
     "identifier" => "SCORE",
      "cardinality" => "single",
      "baseType" => "float",
  ],
]
```

Reply:
Done